vault/router.go

```diff
@@ -522,14 +524,17 @@ func (r *Router) routeCommon(ctx context.Context, req *logical.Request, existenc
 	r.l.RLock()
 	adjustedPath := req.Path
 	mount, raw, ok := r.root.LongestPrefix(ns.Path + adjustedPath)
+	r.logger.Trace("trying to route to mount using adjusted path", "namespace", ns, "adjustedPath", adjustedPath, "mount", mount, "raw", raw, "ok", ok)
 	if !ok && !strings.HasSuffix(adjustedPath, "/") {
 		// Re-check for a backend by appending a slash. This lets "foo" mean
 		// "foo/" at the root level which is almost always what we want.
 		adjustedPath += "/"
 		mount, raw, ok = r.root.LongestPrefix(ns.Path + adjustedPath)
+		r.logger.Trace("after appending / to adjustedPath, trying again to route to mount using adjusted path", "namespace", ns, "adjustedPath", adjustedPath, "mount", mount, "raw", raw, "ok", ok)
 	}
 	r.l.RUnlock()
 	if !ok {
+		r.logger.Trace("route entry not found", "path", req.Path)
```

Comment:
This could also result in a very high log volume.  Can you use a metric instead?

Reply:
I ended up removing this in favor of augmented error messages. I could add additional metrics as well, but after thinking this over a bit, I wasn't sure what kind of metric to add. Are you thinking of a counter each time this failure happens? Or something else?